### ISSUE TITLE
Graduate enableAriaHiddenSiblings flag for modalization

### ISSUE THREAD

**Issue by phkuo** [microsoft/fluentui#19610]
### Environment Information

- **Package version(s)**: 7+
- **Browser and OS versions**: all

### Describe the issue:

This should be a quick but impactful fix.

When a modal popup (like a dialog or callout) appears on the page, all other elements on the page need to be "hidden" from screenreaders. I already wrote the code for this in this PR: https://github.com/microsoft/fluentui/pull/13281
We've turned had it turned on in our product for over a year now without issues. It's time to graduate this to be the default behavior.

`enableAriaHiddenSiblings` should be removed from any current in-progress major release and its behavior turned on by default.

**Comment by ecraig12345** [comment]
Unfortunately at this point we can't remove any props until the next major release of `@fluentui/react`, which if it happens won't be any time soon. We can deprecate it though, and potentially consider changing the default behavior if it's not too disruptive.

**Comment by gouttierre** [comment]
@phkuo - I’ve added this to our backlog for review but it’s low on our priority list as we’re likely not going to get to it immediately due to the fact we’re currently focused on larger coherence work items and enhancements to our performance.

**Comment by phkuo** [comment]
@gouttierre I'd request that this gets looked at soon as it pertains to accessibility. The feature is already done and checked in, all that remains is turning it on by default. We've been using this in SharePoint turned on for a year already.

**Comment by ecraig12345** [comment]
This should probably stay open, but like I mentioned before, we can't remove the prop outside a major version, and changing the default behavior would be risky since that could definitely be considered a breaking behavior change.
